docs/reference/analysis/tokenfilters/multiplexer-tokenfilter.asciidoc

```diff
@@ -0,0 +1,110 @@
+[[analysis-multiplexer-tokenfilter]]
+=== Multiplexer Token Filter
+
+A token filter of type `multiplexer` will emit multiple tokens at the same position,
+each version of the token having been run through a different filter.
+
+
+[float]
+=== Options
+[horizontal]
+filters:: a list of token filters to apply to incoming tokens.  These can be any
+  token filters defined elsewhere in the index mappings, or the special `identity`
+  filter that will emit the token unchanged.  Filters can be chained
+  using a comma-delimited string, so for example `"lowercase, porter_stem"` would
+  apply the `lowercase` filter and then the `porter_stem` filter to a single token
+
+[float]
+=== Settings example
+
+You can set it up like:
+
+[source,js]
+--------------------------------------------------
+PUT /multiplexer_example
+{
+    "settings" : {
+        "analysis" : {
+            "analyzer" : {
+                "my_analyzer" : {
+                    "tokenizer" : "standard",
+                    "filter" : [ "my_multiplexer" ]
+                }
+            },
+            "filter" : {
+                "my_multiplexer" : {
+                    "type" : "multiplexer",
+                    "filters" : [ "identity", "lowercase", "lowercase, porter_stem" ]
```

Comment:
thinking about the API I wonder whether this should be an array of arrays, eg. `[ [], [ "lowercase" ], [ "lowercase", "porter_stem" ] ]`.

Reply:
then maybe make squares optional for single-element lists?
`[ [], "lowercase", [ "lowercase", "porter_stem" ] ]`

Reply:
How easy is that to do with the Settings API?  I've had a quick look, and it seems to expect either single keys or lists of strings, whereas here we'd need a list of variant types.

Reply:
Ok to keep a comma-separated list then. The other thing that worried me is what happens if the user also has a filter whose name is `identity`. Even though I like it a bit less from an API perspective, maybe it would be better to add a `preserve_original` setting or something like that to the `multiplexer` type so that we do not have to add `identity` as a reserved filter name.